opencolorio_config_aces/config/generation/factories.py

```diff
@@ -0,0 +1,570 @@
+
+        view_transform = base_view_transform
+    else:
+        view_transform = ocio.ViewTransform(reference_space)
+
+        if to_reference is not None:
+            view_transform.setTransform(
+                produce_transform(to_reference),
+                ocio.VIEWTRANSFORM_DIR_TO_REFERENCE)
+
+        if from_reference is not None:
+            view_transform.setTransform(
+                produce_transform(from_reference),
+                ocio.VIEWTRANSFORM_DIR_FROM_REFERENCE)
+
+    view_transform.setName(name)
+
+    if family is not None:
+        view_transform.setFamily(family)
+
+    for category in categories:
+        view_transform.addCategory(category)
+
+    if description is not None:
+        view_transform.setDescription(description)
+
+    return view_transform
+
+
+@required('OpenColorIO')
+def look_factory(name,
+                 process_space=None,
+                 description=None,
+                 forward_transform=None,
+                 inverse_transform=None,
+                 base_look=None,
+                 **kwargs):
+    """
+    *OpenColorIO* look factory.
+
+    Parameters
+    ----------
+    name : unicode
+        *OpenColorIO* look name.
+    process_space : unicode, optional
+        *OpenColorIO* look process space, e.g. *OpenColorIO* colorspace or role
+        name.
+    description : unicode, optional
+        *OpenColorIO* look description.
+    forward_transform : dict or object, optional
+        *To Reference* *OpenColorIO* look transform.
+    inverse_transform : dict or object, optional
+        *From Reference* *OpenColorIO* look transform.
+    base_look : dict or ViewTransform, optional
+        *OpenColorIO* base look inherited for initial attribute values.
+
+    Other Parameters
+    ----------------
+    \\**kwargs : dict, optional
+        Keywords arguments.
+
+    Returns
+    -------
+    Look
+        *OpenColorIO* look.
+    """
+
+    import PyOpenColorIO as ocio
+
+    if process_space is None:
+        process_space = ocio.ROLE_SCENE_LINEAR
+
+    if base_look is not None:
+        if isinstance(base_look, Mapping):
+            base_look = look_factory(**base_look)
+
+        look = base_look
+    else:
+        look = ocio.Look()
+
+        look.setProcessSpace(process_space)
+
+        if forward_transform is not None:
+            look.setTransform(produce_transform(forward_transform))
+
+        if inverse_transform is not None:
+            look.setInverseTransform(produce_transform(inverse_transform))
+
+    look.setName(name)
+
+    if description is not None:
+        look.setDescription(description)
+
+    return look
+
+
+@required('OpenColorIO')
+def transform_factory_default(**kwargs):
+    """
+    *OpenColorIO* default transform factory that produces an *OpenColorIO*
+    transform according to given ``name`` keyword argument.
+
+    Other Parameters
+    ----------------
+    name : unicode
```

Comment:
I don't suppose I could convince you guys to rename "name" kwarg used for `transform_factory_default`  to something more descriptive, like "transform_type" or "transform_class"? Compared to how "name" is used with the other factory methods, the term feels kind of overloaded; especially because there's a canonical "name" format metadata attribute that manifests in serialized yaml as an attribute called "name":

`!<LogAffineTransform> {name: Log3G10 to Linear, log_side_slope: 0.224282, lin_side_slope: 155.975327, lin_side_offset: 2.55975327, direction: inverse}`

I was thinking, maybe we could reserve "name", "id", "description", "input_descriptor" and "output_descriptor" for slapping format metadata on freshly-birthed transform instances. It's pretty nice having a way to bundle and deal with the transform parameter data and format metadata at the same points in time and space. 

(I know you guys are already using "name" in a bunch of places, but I figured I should say something now if ever)



feel free to snag this method for certain format metadata to supported transforms (or to group transforms wrapping unsupported transforms):

https://gist.github.com/zachlewis/39a121d07025e2727527d0446b2b5da5

Reply:
Good point, when I looked back then, I did not see any collisions and went for shorter is better but seems like I missed the one highlighted (and possibly others). I'll fix that.

Reply:
Should be fixed in https://github.com/michdolan/OpenColorIO-Config-ACES/pull/2